connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroSerdeSuite.scala

```diff
@@ -49,6 +49,22 @@ class AvroSerdeSuite extends SparkFunSuite {
     }
   }
 
+  test("Test byte conversion") {
+    withFieldMatchType { fieldMatch =>
+      val (top, nest) = fieldMatch match {
+        case BY_NAME => ("foo", "bar")
+        case BY_POSITION => ("NOTfoo", "NOTbar")
+      }
+      val avro = createNestedAvroSchemaWithFields(top, _.optionalInt(nest))
+      val record = new GenericRecordBuilder(avro)
+        .set(top, new GenericRecordBuilder(avro.getField(top).schema()).set(nest, -128).build())
+        .build()
+      val serializer = Serializer.create(CATALYST_STRUCT_WITH_BYTE, avro, fieldMatch)
+      val deserializer = Deserializer.create(CATALYST_STRUCT_WITH_BYTE, avro, fieldMatch)
+      assert(serializer.serialize(deserializer.deserialize(record).get) === record)
```

Comment:
I think you also need another test for that dataframe schema is BYTE type, write to AVRO file and then the read based on the schema with byte type can succeed with expected result.

Reply:
Oh yes I really meant `saveAsTable`. The reason is this test is just test `BYTE` is ser/deser. Then based on the PR description we need to trigger that the preserved avro table schema still contains a byte field.

Maybe this is a right suite for the test https://github.com/apache/spark/blob/master/connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala?